### PR TITLE
mx8 tests

### PR DESCRIPTION
**Pull request by cbrake**

### CODE REVIEW

**Comment by cbrake** [comment]
currently getting:

```
ERROR: Nothing PROVIDES 'yoe-simple-image'
yoe-simple-image was skipped: ['mx8'] overrides cannot be used and need conversion to use the new BSP-specific overrides. Check 'meta-freescale/scripts/convert-bsp-specific-overrides'.

**Comment by cbrake** [comment]
CI: reported this:

ERROR: systemd-1_251.4-r0 do_patch: Applying patch '0020-logind.conf-Set-HandlePowerKey-to-ignore.patch' on target directory '/home/khem/actions-runner-yoe/_work/yoe-distro/yoe-distro/build/tmp/work/cortexa57-yoe-linux/systemd/1_251.4-r0/git'

**Comment by cbrake** [comment]
ok, things are finally parsing and yoe-simple-image build is progressing ...

**Comment by cbrake** [comment]
interesting: freertos-variscite do_fetch ...

**Comment by cbrake** [comment]
next build error:

```
ERROR: imx-atf-2.6+gitAUTOINC+c6a19b1a35-r0 do_compile: oe_runmake failed
ERROR: imx-atf-2.6+gitAUTOINC+c6a19b1a35-r0 do_compile: ExecutionError('/scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115', 1, None, None)
ERROR: Logfile of failure stored in: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/log.do_compile.805115
Log data follows:
| DEBUG: Executing shell function do_compile
| NOTE: make -j 24 CROSS_COMPILE=aarch64-yoe-linux- PLAT=imx8qx LD=aarch64-yoe-linux-ld CC=aarch64-yoe-linux-clang IMX_BOOT_UART_BASE= bl31
| /bin/sh: line 1: aarch64-yoe-linux-clang: command not found
|   CC      bl31/bl31_context_mgmt.c
| make: aarch64-yoe-linux-clang: No such file or directory
| make: *** [Makefile:1253: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/git/build/imx8qx/release/bl31/bl31_context_mgmt.o] Error 127
| make: *** Waiting for unfinished jobs....
|   CC      bl31/bl31_main.c
| make: aarch64-yoe-linux-clang: No such file or directory
| make: *** [Makefile:1253: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/git/build/imx8qx/release/bl31/bl31_main.o] Error 127
| ERROR: oe_runmake failed
| WARNING: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115:171 exit 1 from 'exit 1'
| WARNING: Backtrace (BB generated script):
|       #1: bbfatal_log, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 171
|       #2: die, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 155
|       #3: oe_runmake, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 150
|       #4: do_compile, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 141
|       #5: main, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 184
ERROR: Task (/scratch/treehaven/yoe-distro/sources/meta-freescale/recipes-bsp/imx-atf/imx-atf_2.6.bb:do_compile) failed with exit code '1'


```

**Comment by cbrake** [comment]
imx-atf repo: https://source.codeaurora.org/external/imx/imx-atf/

Trusted Firmware-A (TF-A) is a reference implementation of secure world software
for `Arm A-Profile architectures`_ (Armv8-A and Armv7-A), including an Exception
Level 3 (EL3) `Secure Monitor`_. It provides a suitable starting point for
productization of secure world boot and runtime firmware, in either the AArch32
or AArch64 execution states.

**Comment by kraj** [comment]
> next build error:
> 
> ```
> ERROR: imx-atf-2.6+gitAUTOINC+c6a19b1a35-r0 do_compile: oe_runmake failed
> ERROR: imx-atf-2.6+gitAUTOINC+c6a19b1a35-r0 do_compile: ExecutionError('/scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115', 1, None, None)
> ERROR: Logfile of failure stored in: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/log.do_compile.805115
> Log data follows:
> | DEBUG: Executing shell function do_compile
> | NOTE: make -j 24 CROSS_COMPILE=aarch64-yoe-linux- PLAT=imx8qx LD=aarch64-yoe-linux-ld CC=aarch64-yoe-linux-clang IMX_BOOT_UART_BASE= bl31
> | /bin/sh: line 1: aarch64-yoe-linux-clang: command not found
> |   CC      bl31/bl31_context_mgmt.c
> | make: aarch64-yoe-linux-clang: No such file or directory
> | make: *** [Makefile:1253: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/git/build/imx8qx/release/bl31/bl31_context_mgmt.o] Error 127
> | make: *** Waiting for unfinished jobs....
> |   CC      bl31/bl31_main.c
> | make: aarch64-yoe-linux-clang: No such file or directory
> | make: *** [Makefile:1253: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/git/build/imx8qx/release/bl31/bl31_main.o] Error 127
> | ERROR: oe_runmake failed
> | WARNING: /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115:171 exit 1 from 'exit 1'
> | WARNING: Backtrace (BB generated script):
> |       #1: bbfatal_log, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 171
> |       #2: die, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 155
> |       #3: oe_runmake, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 150
> |       #4: do_compile, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 141
> |       #5: main, /scratch/treehaven/yoe-distro/build/tmp/work/imx8qxp_var_som-yoe-linux/imx-atf/2.6+gitAUTOINC+c6a19b1a35-r0/temp/run.do_compile.805115, line 184
> ERROR: Task (/scratch/treehaven/yoe-distro/sources/meta-freescale/recipes-bsp/imx-atf/imx-atf_2.6.bb:do_compile) failed with exit code '1'
> ```

Add
```
DEPENDS:append:toolchain-clang:class-target = " clang-cross-${TARGET_ARCH} "
```
in the `imx-atf_2.6.bb` recipe.

**Comment by cbrake** [comment]
almost there on `yoe-simple-image` :-)

image

**Comment by kraj** [comment]
> almost there on `yoe-simple-image` :-)
> 
> image

Can you try to find what `IMAGE_FSTYPES` are being generated ?

```
bitbake -e yoe-simple-image | grep "^IMAGE_FSTYPES="
```

**Comment by kraj** [comment]
I have created a living fork for yoedistro activities for variscite BSP layer  [here](https://github.com/yoedistro/meta-variscite-bsp) and its default branch is `yoe/master`,  Lets start using that repository to do the porting work.

**Comment by cbrake** [comment]
something else to fix:

```
[cbrake@ceres yoe-distro]$ . imx8mn-var-som-envsetup.sh 
Setting MACHINE=imx8mn-var-som
/scratch/treehaven/yoe-distro/conf/auto.conf has been updated
Setting DOCKER_REPO to yoedistro/yoe-build:bullseye
[cbrake@ceres yoe-distro]$ bitbake yoe-simple-image
Setting MACHINE=imx8mn-var-som
/scratch/treehaven/yoe-distro/conf/auto.conf has been updated
Setting DOCKER_REPO to yoedistro/yoe-build:bullseye
NOTE: Started PRServer with DBfile: /scratch/treehaven/yoe-distro/cache/prserv.sqlite3, Address: 127.0.0.1:46239, PID: 80
ERROR:  OE-core's config sanity checker detected a potential misconfiguration.
    Either fix the cause of this error or at your own risk disable the checker (see sanity.conf).
    Following is the list of potential problems / advisories:

    Error, the PACKAGE_ARCHS variable (all any noarch ${PACKAGE_EXTRA_ARCHS:tune-cortexa53-crypto} imx8mn_var_som) for DEFAULTTUNE (cortexa53-crypto) does not contain TUNE_PKGARCH (${@bb.utils.contains('TUNE_FEATURES', 'aarch64', '${ARMPKGARCH:tune-cortexa53-crypto}', '${ARMPKGARCH:tune-cortexa53-crypto}', d)}).Toolchain tunings invalid:
Tuning 'cortexa53-crypto' has no defined features, and cannot be used.
```

**Review by kraj — APPROVED** [review]
lgtm

File: sources/meta-yoe/recipes-core/images/machines/imx8qxp-var-som.inc [hunk 1]
```diff
@@ -78,40 +78,4 @@ BOOTFILES = "\
...
 
 do_updater() {
-        echo "IMG_VERSION = ${IMG_VERSION}"
-        rm -rf ${TOPDIR}/deploy/${MACHINE}
-        install -d ${TOPDIR}/deploy/${MACHINE}
-        OLDPWD="$PWD"
-        cd ${TOPDIR}/deploy/${MACHINE}
-        echo "${IMG_VERSION}" > version.txt
-        # rootfs
-        install ${DEPLOY_DIR_IMAGE}/${IMAGE_BASENAME}-${MACHINE}.tar.xz ${ROOTFS_ARTEFACTS}
```
Comment (kraj) [inline sources/meta-yoe/recipes-core/images/machines/imx8qxp-var-som.inc]: or you can use `tar.gz` here instead of `tar.xz`

**Comment by cbrake** [comment]
closing this and continuing work in #715

**Pull request closed without merging**